client/src/components/CurseHome.tsx

```diff
@@ -0,0 +1,190 @@
+  const [allForOneText1, setAllForOneText1] = useState("Cursa All For One este o tură populară, necompetitivă, perfectă pentru cei care vor să facă mișcare în natură pe o distanță accesibilă. Aceasta se adresează tuturor persoanelor, de orice vârstă sau gen, chiar și persoanelor cu dizabilități.");
+  const [allForOneText2, setAllForOneText2] = useState("Tocmai în spiritul caritabil din care s-a și născut Aleargă Pentru Viață, participanții sunt încurajați să ofere drept donație pentru Florin cât consideră.");
+
+  const handleCategoryChange = (category: string) => {
+    setSelectedCategory(category);
+    switch (category) {
+      case 'All for one':
+        setStartTime('Ora de start: 11:30');
+        setDistance('DISTANȚĂ: 400M');
+        setMinimumDonation(
+          'Donatie minima: OPŢIONAL');
+        setAllForOneText1("Cursa All For One este o tură populară, necompetitivă, perfectă pentru cei care vor să facă mișcare în natură pe o distanță accesibilă. Aceasta se adresează tuturor persoanelor, de orice vârstă sau gen, chiar și persoanelor cu dizabilități.");
+        setAllForOneText2("Tocmai în spiritul caritabil din care s-a și născut Aleargă Pentru Viață, participanții sunt încurajați să ofere drept donație pentru Florin cât consideră.");  
+        setMapSrc("https://www.google.com/maps/embed?pb=!1m42!1m12!1m3!1d1051.7564164316468!2d26.050924936618955!3d44.43943916368305!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m27!3e2!4m5!1s0x40b201c187a44aeb%3A0x776c5045f461c6b0!2sBucure%C5%9Fti%20Sectorul%206%2C%20Bucure%C8%99ti%20061344!3m2!1d44.4396542!2d26.052221199999998!4m3!3m2!1d44.439018999999995!2d26.0515156!4m3!3m2!1d44.4397981!2d26.051081699999997!4m3!3m2!1d44.4400312!2d26.0518355!4m3!3m2!1d44.4399332!2d26.0523311!4m3!3m2!1d44.4396554!2d26.052215!5e0!3m2!1sro!2sro!4v1679949014977!5m2!1sro!2sro");
+        break;
+      case 'Copii':
+        setStartTime('Ora de start: 11:10');
+        setDistance('DISTANȚĂ: 400M');
+        setMinimumDonation('Donatie minima: 20 RON');
+        setAllForOneText1("");
+        setAllForOneText2("");
+        setMapSrc("https://www.google.com/maps/embed?pb=!1m42!1m12!1m3!1d1051.7564164316468!2d26.050924936618955!3d44.43943916368305!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m27!3e2!4m5!1s0x40b201c187a44aeb%3A0x776c5045f461c6b0!2sBucure%C5%9Fti%20Sectorul%206%2C%20Bucure%C8%99ti%20061344!3m2!1d44.4396542!2d26.052221199999998!4m3!3m2!1d44.439018999999995!2d26.0515156!4m3!3m2!1d44.4397981!2d26.051081699999997!4m3!3m2!1d44.4400312!2d26.0518355!4m3!3m2!1d44.4399332!2d26.0523311!4m3!3m2!1d44.4396554!2d26.052215!5e0!3m2!1sro!2sro!4v1679949014977!5m2!1sro!2sro");
+        break;
```

Comment:
The component lacks modularity.
I'd like it to be able to receive an external parameter of type JSON with a unique type defined within it.